lib/transactional.pm

```diff
@@ -176,3 +177,28 @@ sub ensure_rollback_service_not_running {
         $output =~ '^(start|running)$' ? sleep 10 : last;
     }
 }
+
+
+# Verify that essential snapshots cannot be deleted, see https://jira.suse.com/browse/SLE-3804
+sub check_undel_snaps {
+    # Create some snapshots with a transactional update snapshot in the middle.
+    record_info "Check undel", "Check that essential snapshots cannot be deleted, creates snapshots and installs a package";
+    assert_script_run("snapper create");
```

Comment:
it would be better to have some message on the snapshot for better tracking.

Reply:
can do, though they are just here to be removed right after. the idea is to simulate a situation where we'd have automatic timelines snapshots created.